icpc-wf/ansible/roles/ssh/tasks/main.yml

```
@@ -0,0 +1,41 @@
+---
+# These tasks configure SSH stuff
+
+- name: Create domjudge .ssh directory
+  file:
+    path: /home/domjudge/.ssh
+    state: directory
+    group: domjudge
+    owner: domjudge
+    mode: 0700
+
+- name: Copy shared SSH private/public key and config to domjudge
+  copy:
+    src: "{{ item }}"
+    dest: /home/domjudge/.ssh/
+    owner: domjudge
+    group: domjudge
+    mode: 0600
+  with_items:
```

I think this can be shorter by removing the repetition over both accounts into 1. If I understand this link correctly: https://docs.ansible.com/ansible/2.4/playbooks_loops.html#nested-loops
```suggestion
  with_nested:
    - [config, id_rsa, id_rsa.pub
    - [root, domjudge]
```

---

I'll try that.

---

Actually, that won't quite work since the ownership is different too.

---

You can use the same variable also for the user+group,
files of root are already implicit from root and now it would become explicit.

---

It also needs to adjust the homedir. Overall, I don't think this makes things simpler, so I prefer to keep this as is.